aea/cli/config.py

```diff
@@ -54,89 +63,269 @@ def get(ctx: Context, json_path: List[str]):
 
-    parent_object_path = json_path[:-1]
-    attribute_name = json_path[-1]
-    parent_object = _get_and_validate_parent_obj(
-        configuration_object, parent_object_path, attribute_name
-    )
+    @property
+    def is_target_agent(self) -> bool:
+        """Is target of get/update is agent config."""
+        return self.component_id is None
 
-    type_ = FROM_STRING_TO_TYPE[type_str]
-    try:
-        if type_ != bool:
-            parent_object[attribute_name] = type_(value)
+    def _load_configuration_object(self) -> Dict:
+        """Load configuration object for component/agent."""
+        if self.is_target_agent:
+            configuration_object = self.agent_config.json
         else:
-            parent_object[attribute_name] = value not in FALSE_EQUIVALENTS
-    except ValueError:  # pragma: no cover
-        raise click.ClickException("Cannot convert {} to type {}".format(value, type_))
+            configuration_object = load_yaml(str(self.path_to_resource_configuration))
 
-    try:
-        configuration_obj = config_loader.configuration_class.from_json(
-            configuration_object
+        self.config_loader.validate(configuration_object)
+        return configuration_object
+
+    def _get_configuration_object_from_agent(self) -> Optional[Dict]:
+        """Get component configuration object from agent component configurations."""
+        if not self.component_id:  # pragma: nocover
+            raise ValueError("component in not set")
+
+        return _try_get_configuration_object_from_aea_config(
+            self.ctx, self.component_id
         )
-        config_loader.validate(configuration_obj.json)
-        config_loader.dump(configuration_obj, open(configuration_file_path, "w"))
-    except Exception:
-        raise click.ClickException("Attribute or value not valid.")
-
-
-def _get_and_validate_parent_obj(
-    conf_obj: Dict, parent_obj_path: List, attr_name: str
-) -> Dict:
-    """
-    Get and validate parent object.
-
-    :param conf_obj: configuration object.
-    :param parent_obj_path: parent object path.
-    :param attr_name: attribute name.
-
-    :return: parent object.
-    :raises: ClickException if attribute is not valid.
-    """
-    try:
-        parent_obj = get_parent_object(conf_obj, parent_obj_path)
-    except ValueError as e:
-        raise click.ClickException(str(e))
-
-    if attr_name not in parent_obj:
-        raise click.ClickException("Attribute '{}' not found.".format(attr_name))
-    if not isinstance(parent_obj.get(attr_name), (str, int, bool, float)):
-        raise click.ClickException(
-            "Attribute '{}' is not of primitive type.".format(attr_name)
+
+    def _get_value_from_configuration_object(self, conf_obj: Dict) -> Any:
+        """Get value from configuration object."""
+        return self._get_parent_object(conf_obj).get(self.attr_name)
+
+    def _get_parent_object(self, conf_obj: Dict) -> Dict:
```

Comment:
dict needs to be fully typed

Reply:
i can type it cause it component configuration and it always differs